src/pages/index.js

```diff
@@ -1,111 +1,227 @@
+    .then((updatedUser) => {
+      const profileImage = document.querySelector(".profile__image");
+      profileImage.src = updatedUser.avatar; // Update profile picture in DOM
+      updateAvatarModal.close();
+    })
+    .catch((err) => console.error("Failed to update avatar:", err))
+    .finally(() => {
+      updateAvatarModal.setLoading(false);
+    });
+});
+
+updateAvatarModal.setEventListeners();
+
+
+// Handle opening the modal when clicking the edit icon
+document.querySelector(".profile__image-container").addEventListener("click", () => {
+  updateAvatarModal.open();
+});
+
+// Ensure form validation
+const updateAvatarFormElement = document.querySelector("#update-avatar-form");
+if (updateAvatarFormElement) {
+  const updateAvatarFormValidation = new FormValidator(config, updateAvatarFormElement);
+  // Enable validation
+  updateAvatarFormValidation.enableValidation();
+  // Disable button initially
+  const saveButton = updateAvatarFormElement.querySelector(".modal__form-button");
+  saveButton.disabled = true;
+  saveButton.classList.add(config.submitButtonDisabled); // Ensure the disabled style is applied
+} else {
+  console.error("Update avatar form not found!");
+}
 
-const addCardForm = document.querySelector(addCardFormSelector);
-const newCardFormValidation = new FormValidator(config, addCardForm);
-newCardFormValidation.enableValidation();
+// Enable form validation
+const editProfileFormElement = document.querySelector('#edit-profile-form');
 
-cardSection.renderItems();
 
-function handleImageClick (cardData) {
-  popupWithImage.open(cardData)
-};
+// Check if forms are valid before initializing validators
+if (editProfileFormElement) {
+  const editProfileValidation = new FormValidator(config, editProfileFormElement);
+  editProfileValidation.enableValidation();
+} else {
+  console.error('Edit profile form not found!');
+}
 
-function handleProfileSubmited (inputValues) {
-  userInfo.setUserInfo({
-    name: inputValues.title,
-    hobby: inputValues.description
-  });
-  editProfileModal.close()
+if (addCardFormElement) {
+  const newCardFormValidation = new FormValidator(config, addCardFormElement);
+  newCardFormValidation.enableValidation();
+} else {
+  console.error('Add card form not found!');
 }
 
-function handleAddSubmitted(inputValues) {
-  const cardData = { name: inputValues.title, link: inputValues.url };
-  const newCard = createCard(cardData);
-  cardSection.addElement(newCard); // Append the card to the card section
-  cardAddModal.close(); // Close the modal
-  addCardFormElement.reset()
 
-}
+// Load initial data
+api.getUserInfo()
+  .then((userData) => {
+    currentUserId = userData._id; // Save current user ID
+    userInfo.setUserInfo({
+      name: userData.name,
+      hobby: userData.about,
+      id: userData._id,
+    });
 
+    const profileImage = document.querySelector(".profile__image");
+    profileImage.src = userData.avatar; // Set the avatar
```

Comment:
`UserInfo` should set the entire profile data (including the **avatar**)

Reply:
I don't understand this comment I am currently setting the data as expected